src/components/Button.js

```diff
@@ -21,13 +25,42 @@ const Button = React.createClass({
 
   getDefaultProps () {
     return {
+      isActive: false,
       primaryColor: StyleConstants.Colors.PRIMARY,
       type: 'primary'
     };
   },
 
+  _renderButtonContent () {
+    const styles = this.styles();
+
+    if (this.props.isActive) {
+      return (
+        <div>
+        <Spin direction='counterclockwise'>
+            <Icon size='20' style={[styles.icon, styles.spinner]} type='spinner' />
+        </Spin>
+          {this.props.actionText ? <div style={styles.actionText}> {this.props.actionText} </div> : null }
+        </div>
+      );
+    } else {
+      return this.props.children;
+    }
+  },
+
```

Comment:
Yes, we're starting to lean toward not making extra render functions if the logic is simple enough to include in the main render function.

So in the main render function you can do something like.

```javascript
{this.props.isActive ? (
  <div>
    <Spin direction='counterclockwise'>
      <Icon size='20' style={[styles.icon, styles.spinner]} type='spinner' />
    </Spin>
    {this.props.actionText ? <div style={styles.actionText}> {this.props.actionText} </div> : null }
  </div>
) : this.props.children)
```